### DISCUSSION TITLE
Is there any way to implement some kind of profiles?

### DISCUSSION THREAD

**Discussion by empresarrollo** [leokhoa/laragon#496] [Q&A]
I have like 3 kind of projects with different needs:

- Very old: with node14, php7.2, mysql5.7
- Old: with node16, php7.4, mysql5.7
- New: with node18, php8.1, mysql8

Is there any way to select some kind of "profile", and change that 3 things at once and reload the services, instead of having to select  each one each time I have to work with that? At least having 3 different .bats?

**Comment by Pen-y-Fan** [comment]
Laragon is portable, it can be set up with different settings and projects per Laragon folder.

**Reply by Pen-y-Fan** [reply]
There isn't a way to switch using a cli:  #412

**Comment by LrkDev** [comment]
Didn't try it but, your best bet right now should be making 3 version of laragon_root%\usr\laragon.ini
Like :
- laragon7.2.ini
- laragon7.4.ini
- laragon8.1.ini
(Easier than editing config file but not impossible)

And automating switching the files and reloading laragon with .bat file located in %laragon_root%\usr\bin\ (as it is added to path by laragon)
- laragon7.2.bat
- laragon7.4.bat
- laragon8.1.bat
(Or one with parameters)

(I tried %laragon_root%\laragon reload but it launch a new instance instead. Did I miss something @Pen-y-Fan ? Using version 6.0)

**Comment by empresarrollo** [comment]
@LrkDev I ended doing just that

I have 3 laragon.ini and 3 bats that change the ini in \usr

I think there is no way to stop all services and start them again. So I have to do it manually. If im working with the 7.4 version, I have to stop the services, exit laragon, and execute the bat for the 8.2 version. 
The bat checks if laragon is running and if it is it shows me a message to close it prior to run laragon again with the new ini

Here is one bat if anyone needs it:

```

@echo off
SETLOCAL EnableExtensions

set EXE=laragon.exe

FOR /F %%x IN ('tasklist /NH /FI "IMAGENAME eq %EXE%"') DO IF %%x == %EXE% goto ProcessFound

goto ProcessNotFound

:ProcessFound

echo %EXE% is running, turn it off firstttt
pause
goto END
:ProcessNotFound

del C:\laragon6\usr\laragon.ini
copy C:\laragon6\lara82.ini C:\laragon6\usr\laragon.ini
start C:\laragon6\laragon.exe


goto END
:END
echo Finished!
```

**Reply by LrkDev** [reply]
Thanks for the feedback @empresarrollo 
Hope @leokhoa will be back and can improve the main app :-)
